tests/zuse/crypto/keccak.hoon

```diff
@@ -0,0 +1,333 @@
+::  https://csrc.nist.gov/projects/cryptographic-standards-and-guidelines/example-values
+::
+::  answer for rawshake-256 was found here:
+::  https://github.com/maandree/libkeccak/blob/master/test.c
+::  for rawshake-128, the implementation was assumed to be correct at the time
+::  these tests were written.
+::
+::
+/+  tester
+=,  keccak:crypto
+::
+=/  keccak-inputs=(map @ud @)
+  =-  (~(run by -) (cury swp 3))
+  %-  ~(gas by *(map @ud @))
+  ^-  (list (pair @ud @))
+  :~
+    :-  0
+    0x0
+  ::
+    :-  1
+    0xcc
+  ::
+    :-  4
+    0xc1ec.fdfc
+  ::
+    :-  8
+    0x4a4f.2024.8451.2526
+  ::
+    :-  64
+    0xe926.ae8b.0af6.e531.76db.ffcc.2a6b.88c6.
+      bd76.5f93.9d3d.178a.9bde.9ef3.aa13.1c61.
+      e31c.1e42.cdfa.f4b4.dcde.579a.37e1.50ef.
+      bef5.555b.4c1c.b404.39d8.35a7.24e2.fae7
+  ::
+    :-  128
+    0x2b6d.b7ce.d866.5ebe.9deb.0802.9521.8426.
+      bdaa.7c6d.a9ad.d208.8932.cdff.baa1.c141.
+      29bc.cdd7.0f36.9efb.1492.8585.8d2b.1d15.
+      5d14.de2f.db68.0a8b.0272.8405.5182.a0ca.
+      e275.234c.c9c9.2863.c1b4.ab66.f304.cf06.
+      21cd.5456.5f5b.ff46.1d3b.461b.d40d.f281.
+      98e3.7325.01b4.860e.add5.03d2.6d6e.6933.
+      8f4e.0456.e9e9.baf3.d827.ae68.5fb1.d817
+  ::
+    :-  255
+      0x3a.3a81.9c48.efde.2ad9.14fb.f00e.18ab.
+      6bc4.f145.13ab.27d0.c178.a188.b614.31e7.
+      f562.3cb6.6b23.3467.75d3.86b5.0e98.2c49.
+      3adb.bfc5.4b9a.3cd3.8338.2336.a1a0.b215.
+      0a15.358f.336d.03ae.18f6.66c7.573d.55c4.
+      fd18.1c29.e6cc.fde6.3ea3.5f0a.df58.85cf.
+      c0a3.d84a.2b2e.4dd2.4496.db78.9e66.3170.
+      cef7.4798.aa1b.bcd4.574e.a0bb.a404.89d7.
+      64b2.f83a.adc6.6b14.8b4a.0cd9.5246.c127.
+      d587.1c4f.1141.8690.a5dd.f012.46a0.c80a.
+      43c7.0088.b618.3639.dcfd.a412.5bd1.13a8.
+      f49e.e23e.d306.faac.576c.3fb0.c1e2.5667.
+      1d81.7fc2.534a.52f5.b439.f72e.424d.e376.
+      f4c5.65cc.a823.07dd.9ef7.6da5.b7c4.eb7e.
+      0851.72e3.2880.7c02.d011.ffbf.3378.5378.
+      d79d.c266.f6a5.be6b.b0e4.a92e.ceeb.aeb1
+  ==
+::
+|_  tst=tester-type:tester
+::
+::  keccak
+::
+++  test-keccak-224
+  ^+  tst
+  =/  answers=(list (pair @ud @))
+    :~
+      :-  0
+      0xf718.3750.2ba8.e108.37bd.d8d3.65ad.b855.
+        9189.5602.fc55.2b48.b739.0abd
+    ::
+      :-  1
+      0xa9ca.b59e.b40a.10b2.4629.0f2d.6086.e32e.
+        3689.faf1.d26b.470c.899f.2802
+    ::
+      :-  4
+      0xe405.869d.a146.4a70.5700.a3cb.ce13.1aab.
+        eeba.9c8d.2fe6.576b.21bc.be16
+    ::
+      :-  8
+      0x7a5c.2cb3.f999.dd00.eff7.3999.6331.4ca6.
+        47dd.0e5a.e1bd.dec6.11f8.338d
+    ::
+      :-  64
+      0xc533.dcf8.8cd1.a5df.f22b.914d.3875.bd57.
+        fc17.b2e1.f474.ae36.0c38.77d2
+    ::
+      :-  128
+      0xaf3e.0cc6.e645.01f1.0fd3.9722.e852.355f.
+        d6d8.0d32.1906.31e2.f06c.22ad
+    ::
+      :-  255
+      0x5af5.6987.ea9c.f11f.cd0e.ac5e.bc14.b037.
+        365e.9b11.23e3.1cb2.dfc7.929a
+    ==
+  |-
+  ?~  answers  tst
+  =.  tst
+    %^  expect-eq.tst
```

Comment:
I think this would be easier to read with aliases for `p.i.answers` and `q.i.answers`.